src/components/WorkspaceSelectionDialog.js

```diff
@@ -2,55 +2,127 @@ import React, { Component } from 'react';
 import Dialog from '@material-ui/core/Dialog';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogTitle from '@material-ui/core/DialogTitle';
-import FormControl from '@material-ui/core/FormControl';
-import InputLabel from '@material-ui/core/InputLabel';
-import Select from '@material-ui/core/Select';
-import MenuItem from '@material-ui/core/MenuItem';
+import { List, ListItem, ListItemText } from '@material-ui/core';
 import PropTypes from 'prop-types';
+import { keys, chars } from '../lib/KeyHelper';
 
 /**
  */
 export class WorkspaceSelectionDialog extends Component {
+  /** */
+  constructor(props) {
+    super(props);
+    const { workspaceType } = this.props;
+    this.state = {
+      selected: workspaceType,
+    };
+
+    this.selectPreviousItem = this.selectPreviousItem.bind(this);
+    this.selectNextItem = this.selectNextItem.bind(this);
+    this.keyDownHandler = this.keyDownHandler.bind(this);
+    this.workspaceTypes = ['elastic', 'mosaic'];
+  }
+
+  /**
+   * Propagate workspace type selection into the global state
+   */
+  handleworkspaceTypeChange(workspaceType) {
+    const { handleClose, updateConfig } = this.props;
+
+    updateConfig({
+      workspace: {
+        type: workspaceType,
+      },
+    });
+    handleClose();
+  }
+
+  /** */
+  selectPreviousItem() {
```

Comment:
A simple toggling function would have been sufficient here, but I tried to be prepared for some additional workspaceTypes that may come in the future.